ntoskrnl/ps/process.c

```diff
@@ -1614,4 +1677,234 @@ NtOpenProcess(OUT PHANDLE ProcessHandle,
     /* Return status */
     return Status;
 }
+
+PEPROCESS NtpGetAdjacentProcess(const BOOLEAN is_backwards, const PEPROCESS current)
+{
+    return !is_backwards
+               ? PsGetNextProcess(current)
+               : PsGetPreviousProcess(current);
+}
+
+NTSTATUS
+NTAPI
+NtGetNextProcess(
+    _In_ HANDLE ProcessHandle,
+    _In_ ACCESS_MASK DesiredAccess,
+    _In_ ULONG HandleAttributes,
+    _In_ ULONG Flags,
+    _Out_ PHANDLE NewProcessHandle
+)
+{
+    const KPROCESSOR_MODE previousMode = KeGetPreviousMode();
```

Comment:
"PreviousMode", no need to be const.

Reply:
Regarding `const`, is it not a good thing that adds compile-time checks and increases code safety? Google codestyle [recommends](https://google.github.io/styleguide/cppguide.html#Use_of_const) it for local variables (and for some C++ uses).

Reply:
Our code generally doesn't do this, but I wouldn't complain about new code that does.
However it's misleading on parameters so I would discourage it there (like in NtpGetAdjacentProcess above).